### ISSUE TITLE
[Yandex Search API] Some searches are giving infinite response time

### ISSUE THREAD

**Issue by kagermanov27** [serpapi/public-roadmap#687]
<img width="1144" alt="image" src="https://user-images.githubusercontent.com/73674035/222272512-20a007d9-9697-4cbc-877c-949788d35c14.png">

[Inspect](https://serpapi.com/searches/63fea7d817f923239413fa88/inspect)
[Playground](https://serpapi.com/search.json?&engine=yandex&lr=11316&no_cache=true&text=Coffee&yandex_domain=yandex.com)

Urgent because it has the potential to bring down deployment on some servers.

**Comment by kagermanov27** [comment]
I was planning to do a separate PR to ease this. But looking into the issue, I have realized we only need to prevent lr parameter only for yandex.com now. All the other ones are working in https://github.com/serpapi/SerpApi/pull/3615.

So I have made a little addition there. Once that PR is merged, we won't have big response times for all domains.

Of course, the root cause for the main issue, the search running for 20h is still ambiguous.

**Comment by kagermanov27** [comment]
I have accidentally recreated it on my localhost for some searches:
<img width="1465" alt="image" src="https://user-images.githubusercontent.com/73674035/222408046-472cf321-cd33-491b-8b64-ce3ab59a2c47.png">
<img width="1465" alt="image" src="https://user-images.githubusercontent.com/73674035/222408100-7c97d21e-c525-44ce-bd5d-fdfc13ac323f.png">
<img width="1465" alt="image" src="https://user-images.githubusercontent.com/73674035/222408129-50772fc9-ca15-47ef-8e38-d55d69f3d9c7.png">

The problem seems to be actual and unrelated to lr parameter.

**Comment by kagermanov27** [comment]
Another example on Google Local Services

[Inspect](https://serpapi.com/searches/64bfd0dbf0adfb8cee31c552/inspect) (Reported by @marm123)
